### PR TITLE
feat(update): Use metadata endpoint, support BigQuery

### PR DESCRIPTION
**Pull request by spencerwilson**
- `dpm update`: Use metadata endpoint, support BigQuery
- Delete dead Snowflake code and remove associated `tonic` dep

## Test plan

Verified that `dpm update`:

- Carries PKs over from input descriptor
- Correctly identifies "no changes" scenario

### CODE REVIEW

**Comment by spencerwilson** [comment]
Current dependencies on/for this PR:
* main
  * **PR #192** <a href="https://app.graphite.dev/github/pr/patch-tech/dpm/192?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #193** <a href="https://app.graphite.dev/github/pr/patch-tech/dpm/193?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #194** <a href="https://app.graphite.dev/github/pr/patch-tech/dpm/194?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #195** <a href="https://app.graphite.dev/github/pr/patch-tech/dpm/195?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/patch-tech/dpm/195?utm_source=stack-comment).

File: src/command/init.rs [hunk 1]
```diff
@@ -60,6 +79,43 @@ pub async fn init(
...
+    let allow_list = refinement.map(|r| r.into_allow_list());
+    let tables_for_prompt = tables_from_metadata(response, allow_list.as_ref())?;
+
+    let selected_tables = select_tables_and_keys(tables_for_prompt)?;
+
+    let descriptor = DataPackage {
+        id: uuid7(),
+        name: package_name.to_owned(),
+        description: None,
+        version: "0.1.0".parse().unwrap(),
```
Comment (ajmasci) [inline src/command/init.rs:91]: nit: add a comment indicating that the version is a placeholder?

Comment (ajmasci) [reply src/command/init.rs:91]: Ah, yes. I forgot that this was in the `init` function.

File: src/util.rs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+pub enum AllowListItem {
+    BigQueryTable(String),
```
Comment (ajmasci) [inline src/util.rs:2]: As discussed IRL, please add some comments here for readability.

**Review by ajmasci — APPROVED** [review]
:ship it:

**Pull request merged**